From a3a7d0a359b046f739f624a54af88b05724a33bb Mon Sep 17 00:00:00 2001
From: Zidan Wang <zidan.wang@freescale.com>
Date: Tue, 1 Sep 2015 19:11:50 +0800
Subject: [PATCH 0655/1691] MLK-11480-03 ARM: dts: add spdif sound card support
 for imx6sx-sdb board

commit a3a7d0a359b046f739f624a54af88b05724a33bb from
git://git.freescale.com/imx/linux-2.6-imx.git imx_4.1.15_1.0.0_ga

add spdif sound card support for imx6sx-sdb board

Signed-off-by: Zidan Wang <zidan.wang@freescale.com>
---
 arch/arm/boot/dts/imx6sx-sdb.dtsi | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/arch/arm/boot/dts/imx6sx-sdb.dtsi b/arch/arm/boot/dts/imx6sx-sdb.dtsi
index b8988f2..a2dcb47 100644
--- a/arch/arm/boot/dts/imx6sx-sdb.dtsi
+++ b/arch/arm/boot/dts/imx6sx-sdb.dtsi
@@ -155,6 +155,14 @@
 		mux-ext-port = <6>;
 		hp-det-gpios = <&gpio1 17 1>;
 	};
+
+	sound-spdif {
+		compatible = "fsl,imx-audio-spdif",
+			   "fsl,imx6sx-sdb-spdif";
+		model = "imx-spdif";
+		spdif-controller = <&spdif>;
+		spdif-out;
+	};
 };
 
 &adc1 {
@@ -301,6 +309,12 @@
 	status = "okay";
 };
 
+&spdif {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_spdif>;
+	status = "okay";
+};
+
 &ssi2 {
 	status = "okay";
 };
@@ -535,6 +549,12 @@
                         >;
                 };
 
+		pinctrl_spdif: spdifgrp {
+			fsl,pins = <
+				MX6SX_PAD_SD4_DATA4__SPDIF_OUT          0x1b0b0
+			>;
+		};
+
 		pinctrl_vcc_sd3: vccsd3grp {
 			fsl,pins = <
 				MX6SX_PAD_KEY_COL1__GPIO2_IO_11		0x17059
-- 
1.9.1

